File: 058-length-of-last-word/Solution.java
class Solution58 {
    public int lengthOfLastWord(String s) {

        //宣告計數器
        int Ans = 0;
        //進入迴圈判定
        for (int i = s.length()-1; i>=0 ; i--){

            System.out.println(i);
            if (s.charAt(i) == ' ' && Ans == 0){
                System.out.println("in if");
                continue;
            }
            else if (s.charAt(i) == ' ' && Ans != 0){
                System.out.println("in elif");
                return Ans;
            }
            else {
                Ans += 1;
                System.out.println("in else" + Ans);
            }
        }
        return Ans;
    }

    static void main(String[] args){
        Solution58 sol = new Solution58();
        String s1 = "emoon  ";
        System.out.println(sol.lengthOfLastWord(s1));
    }
}

File: 058-length-of-last-word/solution.py
class Solution1122:
    def lengthOfLastWord(self, s: str) -> int:

        #宣告計數器
        Ans = 0

        #進入迴圈判定
        for i in range (len(s)-1, -1, -1):
            print(i)
            
            if s[i] == " " and Ans == 0:
                print("in if")
                continue

            elif s[i] == " " and Ans != 0:
                print("in elif")
                return Ans
            
            else:
                Ans += 1
                print("in else", Ans)
        
        return Ans

result = Solution1122().lengthOfLastWord("y")
print(result)